test/constraint.jl

```diff
@@ -122,6 +122,29 @@ function test_delete_constraints(ModelType, ::Any)
     @test_throws Exception JuMP.delete(second_model, constraint_ref)
 end
 
+function test_delete_constraints_name(ModelType, ::Any)
+    model = ModelType()
+    @variable(model, x)
+    @constraint(model, con_ref, 2x <= 1)
+    @test JuMP.is_valid(model, con_ref)
+    @test model[:con_ref] === con_ref
+    JuMP.delete(model, :con_ref)
+    @test !JuMP.is_valid(model, con_ref)
+    @test_throws KeyError(:con_ref) model[:con_ref]
+    @constraint(model, con_ref[i=2:3], 3x >= i)
+    for i in 2:3
+        @test JuMP.is_valid(model, con_ref[i])
+        con = constraint_object(con_ref[i])
+        @test isequal_canonical(jump_function(con), 3x)
```

Comment:
Why is this loop needed? Why are we worried about `constraint_object` returning the wrong value?

Reply:
It's more for `is_valid`